File: Medium/Java/807. Max Increase to Keep City Skyline.java
class Solution {
    public int maxIncreaseKeepingSkyline(int[][] grid) {
        int n = grid.length;
        int sum = 0;
        int[] row = new int[n];
        int[] col = new int[n];

        for(int i = 0; i < n; i++){
            row[i] = grid[i][0];
            col[i] = grid[0][i];
            for(int j = 0; j < n; j++){
                row[i] = Math.max(row[i], grid[i][j]);
                col[i] = Math.max(col[i], grid[j][i]);
            }
        }

        for(int i = 0; i < n; i++){
            for(int j = 0; j < n; j++){
                sum += Math.min(row[i], col[j]) - grid[i][j];
            }
        }

        return sum;
    }
}

File: Medium/JavaScript/807. Max Increase to Keep City Skyline.js
var maxIncreaseKeepingSkyline = function(grid) {
    let n = grid.length;
    let sum = 0;
    let row = Array(n).fill(0);
    let col = Array(n).fill(0);

    for(let i = 0; i < n; i++){
        row[i] = grid[i][0];
        col[i] = grid[0][i];
        for(let j = 0; j < n; j++){
            row[i] = Math.max(row[i], grid[i][j]);
            col[i] = Math.max(col[i], grid[j][i]);
        }
    }

    for(let i = 0; i < n; i++){
        for(let j = 0; j < n; j++){
            sum += Math.min(row[i], col[j]) - grid[i][j];
        }
    }

    return sum;
};
